content/posts/manage-list-feature-flags-with-file/index.md

```diff
@@ -0,0 +1,229 @@
++++
+date = "2021-12-29T14:28:23+07:00"
+author = "nhattien.ho"
+description = "Manage list feature flags with file yaml/json."
+title = "Create feature flags using file in Unleash"
+categories = ["Feature toggle", "Testing"]
+tags = ["Feature toggle", "Unleash", "Node.js"]
+slug = "manage-list-flags-with-file"
++++
+
+In this blog, we will learn how to create a list of feature toggles in Unleash using file. First, let's learn some concept of feature toggle and Unleash 
+#### What is feature toggle (feature flag)?
+Feature toggle (Feature flag) is a technique that allows you to disable some functionality of your application, through settings, without having to deploy new code.
+#### What is Unleash?
+[Unleash](https://docs.getunleash.io/) is a feature toggle system, that gives you a great overview of all feature toggles across all your applications and services. It comes with official client implementations for Java, Node.js, Go, Ruby, Python and .Net.
+#### Why do we use file to create feature flags instead of Admin UI?
+In Manabie, we manage the addition and removal of feature flags with pull request, so we need a solution to manage the list of feature flags by file. Developers can see and change the list of feature flags by modifying this file, don't need to access the admin page. The list of feature flags will be updated after we deploy new code backend.
+#### How to use file to manage list of feature flags in Unleash?
+To run Unleash locally, we need:
+* [Node.js](https://nodejs.org/en/download/) (version 14 or later)
+* [PostgreSQL](https://www.postgresql.org/download/) (version 10 or later)
+* [Create an unleash user and database.](https://docs.getunleash.io/deploy/database-setup)
+##### Start Unleash server
+* Create a new folder/directory on your development computer.
+* From a terminal/bash shell, install the dependencies:
+``` bash
+ npm init
+ npm install unleash-server --save
+```
+* Create a file called server.js, paste the following into it and save.
+```javascript
+const unleash = require('unleash-server');
+let options = {
+    db: {
+        ssl: false,
+        host: 'localhost',
+        port: 9432,
+        database: 'unleash',
+        user: 'root',
+        password: 'root',
+    },
+    server: {
+        port: 4242,
+    },
+}
+unleash
+  .start(options)
+  .then((unleash) => {
+    console.log(
+      `Unleash started on http://localhost:${unleash.app.get('port')}`,
+    );
+  });
+```
+* Run `server.js`:
+```bash
+node server.js
+```
+Once the Unleash server has started, you will see the message:
+```bash
+Unleash started on http://localhost:4242
+```
+The first time Unleash starts it will create a default user which you can use to sign-in to you Unleash instance and add more users with:
```

Comment:
The first time Unleash starts, it will create a default user which you can use to sign-in to you Unleash instance and add more users with: